addon/record-data.js

```diff
@@ -146,7 +146,9 @@ export default class M3RecordData {
     this.storeWrapper = storeWrapper;
     if (CUSTOM_MODEL_CLASS) {
       this.globalM3CacheRD = globalM3CacheRD;
-      this.globalM3CacheRD[this.id] = this;
+      if (!baseRecordData && !parentRecordData && this.id) {
```

Comment:
* avoid the boolean coercion here
* use the passed in `id` variable (to avoid reading from the class prototype during construction)
* why would `id` be falsey here?

```suggestion
      if (baseRecordData !== undefined && parentRecordData !== undefined && id !== undefined) {
```